modal/_container_entrypoint.py

```diff
@@ -61,20 +61,24 @@ class UserException(Exception):
     pass
 
 
-def run_with_signal_handler(coro):
-    """Execute coro in an event loop, with a signal handler that cancels
-    the task in the case of SIGINT or SIGTERM. Prevents stray cancellation errors
-    from propagating up."""
-
-    loop = asyncio.new_event_loop()
-    task = asyncio.ensure_future(coro, loop=loop)
-    for s in [signal.SIGINT, signal.SIGTERM]:
-        loop.add_signal_handler(s, task.cancel)
-    try:
-        result = loop.run_until_complete(task)
-    finally:
-        loop.close()
-    return result
+class EventLoop:
+    """Manage an event loop for executing coroutines while handling SIGINT/SIGTERM.
+
+    Prevents stray cancellation errors from propagating up.
+    """
+
+    def __enter__(self):
+        self.loop = asyncio.new_event_loop()
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.loop.close()
+
+    def run(self, coro):
+        task = asyncio.ensure_future(coro, loop=self.loop)
+        for s in [signal.SIGINT, signal.SIGTERM]:
+            self.loop.add_signal_handler(s, task.cancel)
```

Comment:
Not totally clear to me whether, now that we're going to run multiple coroutines, if we should remove existing signal handlers when this method is called for the second+ time.